osu.Game/Beatmaps/BeatmapInfo.cs

```diff
@@ -169,6 +170,10 @@ public BeatmapDifficulty BaseDifficulty
         [Ignored]
         public APIBeatmap? OnlineInfo { get; set; }
 
+        /// <summary>
+        /// The maximum achievable combo on this beatmap, populated for online info purposes only.
+        /// Todo: This should never be used nor exist, but is still relied on in <see cref="ScoresContainer.Scores"/> since <see cref="IBeatmapInfo"/> can't be used yet.
```

Comment:
Not sure this is loud enough. I'd rather straight up obsolete this and suppress locally in the `ScoresContainer` usage.

Reply:
Sure, didn't think about obsoleting since it won't be eventually removed, but seems like another use case for it.

Reply:
I don't know that it won't be removed? As we have things max combo depends on mods. So I don't know that this property can ever be useful again.

Reply:
I meant more like it won't be removed after a specific date in time (i.e. there are still usages to it that need to be updated), sorry.

Reply:
I'd hope to see it gone sooner rather than later, so I'm personally fine with adding urgency to it via a straight up obsoletion.